retroshare-gui/src/gui/Posted/PhotoView.cpp

```diff
@@ -0,0 +1,106 @@
+/*******************************************************************************
+ * retroshare-gui/src/gui/Posted/PhotoView.cpp                                *
+ *                                                                             *
+ * Copyright (C) 2020 by RetroShare Team       <retroshare.project@gmail.com>  *
+ *                                                                             *
+ * This program is free software: you can redistribute it and/or modify        *
+ * it under the terms of the GNU Affero General Public License as              *
+ * published by the Free Software Foundation, either version 3 of the          *
+ * License, or (at your option) any later version.                             *
+ *                                                                             *
+ * This program is distributed in the hope that it will be useful,             *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of              *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                *
+ * GNU Affero General Public License for more details.                         *
+ *                                                                             *
+ * You should have received a copy of the GNU Affero General Public License    *
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.       *
+ *                                                                             *
+ *******************************************************************************/
+
+#include "PhotoView.h"
+
+#include <QMenu>
+#include <QFileDialog>
+#include <QMessageBox>
+
+#include "gui/gxs/GxsIdDetails.h"
+#include "gui/RetroShareLink.h"
+
+#include <retroshare/rsidentity.h>
+#include <retroshare/rsposted.h>
+
+/** Constructor */
+PhotoView::PhotoView(QWidget *parent)
+: QDialog(parent),
+	ui(new Ui::PhotoView)
+{
+  /* Invoke the Qt Designer generated object setup routine */
+  ui->setupUi(this);
+
+  setAttribute(Qt::WA_DeleteOnClose, true);
+  
+  connect(ui->shareButton, SIGNAL(clicked()), this, SLOT(copyMessageLink()));
+}
+
+/** Destructor */
+PhotoView::~PhotoView()
+{
+	delete ui;
+}
+
+void PhotoView::setPixmap(const QPixmap& pixmap) 
+{
+	QPixmap sqpixmap = pixmap.scaled(640,480, Qt::KeepAspectRatio, Qt::SmoothTransformation);
+
+	if(pixmap.width() > 800 || pixmap.height() > 700)
```

Comment:
Additional functionality, resize the image if the window is resized.
https://stackoverflow.com/a/36813472

Reply:
When i set 640*480
The pictures are to small, and often sometime the pictures which we sharing it can be over 800*600 
I tested it lots of, and when i resize 640*480 pictures again to 640*480
It displays smaller on PhotoView then in the PostedItem & Card View.

I see a effect that the comics with 640*480 which are again resized its blurred the text of the pictures.
I wanted skip resize when picture is under dimensions 800*600, to avoid blurring the picture

Reply:
> Additional functionality, resize the image if the window is resized.
> https://stackoverflow.com/a/36813472

for that i think i need more time dont know if i can do it this week

Reply:
> Additional functionality, resize the image if the window is resized.
> https://stackoverflow.com/a/36813472

here i found similiar idea, need time to implement that for the PhotoViews Label

https://stackoverflow.com/questions/8211982/qt-resizing-a-qlabel-containing-a-qpixmap-while-keeping-its-aspect-ratio/22618496#22618496